tests/spec/fs.rename.spec.js

```diff
@@ -151,4 +151,44 @@ describe('fs.rename', function() {
       });
     });
   });
+
+  it('(promise version) should rename an existing file', function(done) {
+    var complete1 = false;
+    var complete2 = false;
+    var fs = util.fs();
+
+    function maybeDone() {
+      if(complete1 && complete2) {
+        done();
+      }
+    }
+
+    fs.open('/myfile', 'w+', function(error, fd) {
+      if(error) throw error;
+
+      fs.close(fd, function(error) {
+        if(error) throw error;
+
+        fs.promises.rename('/myfile', '/myotherfile').then(
```

Comment:
I think it's confusing that you're mixing callbacks with promises in this test.  Since you're testing the promise version, can you switch all these over to use promises instead?

Reply:
Replaced all async functions with their promise equivalents (commit:  5abefb6c6a467eb0271caddc0685760427fd57bc). I also noticed later that my IDE added some spaces after all occurrences of "function" and "if" so this was corrected in 98611f4d7669488d365e8f07f1b43f582a52da6f